test/components/views/rooms/wysiwyg_composer/components/LinkModal-test.tsx

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { FormattingFunctions } from "@matrix-org/matrix-wysiwyg";
+import { render, screen, waitFor } from "@testing-library/react";
+import React from "react";
+import userEvent from "@testing-library/user-event";
+
+import { LinkModal } from "../../../../../../src/components/views/rooms/wysiwyg_composer/components/LinkModal";
+import { mockPlatformPeg } from "../../../../../test-utils";
+import * as selection from "../../../../../../src/components/views/rooms/wysiwyg_composer/utils/selection";
+import { SubSelection } from "../../../../../../src/components/views/rooms/wysiwyg_composer/types";
+
+describe("LinkModal", () => {
+    const composer = {
+        link: jest.fn(),
+    } as unknown as FormattingFunctions;
```

Comment:
Storing `FormattingFunction` in a `composer` variable all over the place is a bit confusing to me. Why is that so?

Also is it possible to avoid the cast here?

Reply:
I have renamed the variable. Casting here make the test declaration easier.